### PR TITLE
Db

### PR DESCRIPTION
**Pull request by Madeeha-Anjum**
This PR successfully creates a user(rider/driver) in the database and returns a user(rider/driver) object and assigns it to the section user. 
I also implemented enum. 
Created a todo for *UI guys* in CreateAccountActivity. 
Updated method names. 
Complete part of this card: https://trello.com/c/JMA9cluw/25-firebase-setup   |||AuthManger: CreateUserWithEmailAndPassword() -> FireBaseUser
GetCurrentUser() -> FireBaseUser |||| is complete. 
Please delete my branch after/if merged.
I also just used a gitnore from this website http://www.gitignore.io/ not sure if luke updated but yeah?

### CODE REVIEW

File: .idea/workspace.xml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
   <component name="ChangeListManager">
+<<<<<<< HEAD
```
Comment (evantimms) [inline .idea/workspace.xml]: Resolve this merge conflict

File: .idea/workspace.xml [hunk 2]
```diff
@@ -21,6 +24,7 @@
       <change afterPath="$PROJECT_DIR$/application/app/src/main/java/com/example/buber/Services/ApplicationService.java" afterDir="false" />
       <change beforePath="$PROJECT_DIR$/application/.idea/vcs.xml" beforeDir="false" afterPath="$PROJECT_DIR$/application/.idea/vcs.xml" afterDir="false" />
     </list>
+>>>>>>> 2ef7666997348c1333d377e66bf634fe37529f70
```
Comment (evantimms) [inline .idea/workspace.xml]: Same here

File: application/app/src/main/java/com/example/buber/DB/AuthDBManager.java [hunk 3]
```diff
@@ -54,32 +55,38 @@ public void signOut() {
      * @param password
      * New User's entered email and password
      */
-    public void createAccount(String email, String password){
+    //Note to self: this "createUserWithEmailAndPassword(email, password)" query takes time so on success will take time
```
Comment (evantimms) [inline application/app/src/main/java/com/example/buber/DB/AuthDBManager.java:83]: We will resolve this issue in the future by running queries in separate threads, but for authentication its probably fine to just wait for the query to finish and then update the UI (the user wont be doing anything during this time anyways)

File: application/app/src/main/java/com/example/buber/Services/DatabaseManager.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
+package com.example.buber.Services;
```
Comment (evantimms) [inline application/app/src/main/java/com/example/buber/Services/DatabaseManager.java]: Why do we have a DatabaseManger class and a DBManger class?

File: application/app/src/main/java/com/example/buber/DB/DBManager.java [hunk 5]
```diff
@@ -14,46 +11,52 @@
     private FirebaseFirestore database;      // Database connection
 
     private CollectionReference collectionDriver, collectionRider, collectionTrip;
-    private ArrayList<Driver> dataDriver; //Array of class objects of Driver
-    private ArrayList<Rider> dataRider; //Array of class objects of Rider
-    private ArrayList<Trip> dataTrip; //Array of class objects of Trips
+//    private ArrayList<Driver> dataDriver; //Array of class objects of Driver
```
Comment (evantimms) [inline application/app/src/main/java/com/example/buber/DB/DBManager.java]: Just delete this commented code

File: application/app/src/main/java/com/example/buber/DB/AuthDBManager.java [hunk 6]
```diff
@@ -54,32 +55,38 @@ public void signOut() {
...
     public FirebaseUser getcurrentUser(){
         return  mAuth.getCurrentUser();
     }
+
     /**
-     * This method is specifically  for entering the UserID's into rider or driver class
+     * This method is specifically  for entering the DocID's into rider or driver DocID
      */
-    public String getcurrentUserID(){
+    public String getcurrentUserDocID(){
```
Comment (evantimms) [inline application/app/src/main/java/com/example/buber/DB/AuthDBManager.java]: Nice

File: application/app/src/main/java/com/example/buber/Model/Account.java [hunk 7]
```diff
@@ -4,13 +4,13 @@
     private String firstName;
     private String lastName;
     private String email;
-    private String password;
+//    private String password;
```
Comment (evantimms) [inline application/app/src/main/java/com/example/buber/Model/Account.java]: delete

File: application/app/src/main/java/com/example/buber/Model/Account.java [hunk 8]
```diff
@@ -4,13 +4,13 @@
...
 
     public Account(String firstName, String lastName, String email, String password) {
         this.firstName = firstName;
         this.lastName = lastName;
         this.email = email;
-        this.password = password;
+//        this.password = password;
```
Comment (evantimms) [inline application/app/src/main/java/com/example/buber/Model/Account.java]: delete

File: application/app/src/main/java/com/example/buber/Model/Account.java [hunk 9]
```diff
@@ -37,11 +37,11 @@ public void setEmail(String email) {
...
     }
 
-    public String getPassword() {
-        return password;
-    }
-
-    public void setPassword(String password) {
-        this.password = password;
-    }
+//    public String getPassword() {
```
Comment (evantimms) [inline application/app/src/main/java/com/example/buber/Model/Account.java]: Don't worry about deleted commented code, if you don't need it its just better to get rid of it.

File: application/app/src/main/java/com/example/buber/Model/Contact.java [hunk 10]
```diff
@@ -2,5 +2,9 @@
 
 interface Contact {
 
+
+    //Dose this have attributes?
```
Comment (evantimms) [inline application/app/src/main/java/com/example/buber/Model/Contact.java]: Not fully implemented yet

File: application/app/src/main/java/com/example/buber/Services/DatabaseManager.java (hunk 4 above)
Comment (evantimms) [reply application/app/src/main/java/com/example/buber/Services/DatabaseManager.java]: This seems duplicated, seems like we can get away with just our DBManger class so we should delete this one

File: application/app/src/main/java/com/example/buber/Services/ApplicationService.java [hunk 11]
```diff
@@ -21,4 +26,56 @@ public static User createNewUser(
             String phoneNumber
     ) { return null; }
 
+
+//    --------------------MADEEHA---------------------------------------
+    /*TODO:
+       //get username and password from UI when user creates an account
+       //append @Uber.com to the username
```
Comment (evantimms) [inline application/app/src/main/java/com/example/buber/Services/ApplicationService.java]: Why are we appending @Uber.com?

File: application/app/src/main/java/com/example/buber/DB/DBManager.java [hunk 12]
```diff
@@ -14,46 +11,52 @@
...
+//        dataTrip = new ArrayList<>();
     }
 
     // TODO: Implement Create, Read, ReadAll, Update and Delete for any document in the Driver,
     // Rider and Trip Collection
-
-    // TODO: Remove populate -> replace w/ getAll
-    public void populateAllTrips() {
-        //ToDo: List of all trips
+// TODO: Remove populate -> replace w/ getAll
```
Comment (evantimms) [inline application/app/src/main/java/com/example/buber/DB/DBManager.java]: Remove this commented code, and lets create a get and getAll method for our collections (pass a docId and return the document). Luke will need this functionality.

File: application/app/src/main/java/com/example/buber/Services/ApplicationServiceHelper.java [hunk 13]
```diff
@@ -0,0 +1,9 @@
+package com.example.buber.Services;
+
+import com.example.buber.Model.User;
+
+public interface ApplicationServiceHelper {
```
Comment (evantimms) [inline application/app/src/main/java/com/example/buber/Services/ApplicationServiceHelper.java]: What is your rational for using this interface? Not saying its wrong just curious. I like how you pass the reference to the Service class in the callback, good thinking on that one.

File: application/app/src/main/java/com/example/buber/Services/ApplicationService.java [hunk 14]
```diff
@@ -21,4 +26,56 @@ public static User createNewUser(
...
+       //call my create account function in LoginAuth:              LoginAuth.createAccount(username@uber.com, password, (method called on sucess))
```
Comment (evantimms) [inline application/app/src/main/java/com/example/buber/Services/ApplicationService.java]: This is great, but lets just worry about getting a User object. Next week / sprint we can think a little more about how to do driver/rider stuff. So right now I think we need the service methods. 
1. SignIn/Login
- takes a username and password, calls ur AuthManger login functionality and gets the docId for the user
- calls ur DB.get(docId) for the users collection and returns that user
2. Logout
- takes a username and password, calls ur DB logout function and returns null
3. CreateNewUser
- takes email, id? and creates a new FireBase user using the AuthManger createAccount function
- Return the FireBaseUser docId, create a new User class (using rest of parameters passed down from UI)
- Take new User and stick in User collections using the DB.create user function

Comment (evantimms) [reply application/app/src/main/java/com/example/buber/Services/ApplicationService.java]: It is okay if we have lots of null fields in the User, I am literally just concerned about making sure we can log in / log out / create an account. We can edit the User class so it only takes the attributes Mike and Nick defined in the UI.

File: application/app/src/main/java/com/example/buber/Services/ApplicationService.java [hunk 15]
```diff
@@ -21,4 +26,56 @@ public static User createNewUser(
...
+        login.createAccount("madeeha@BUber.com", "123456" , this);    // If account is created aftersuccessfulCreataAccount() is called
+    }
+
+
+    public User aftersuccessfulCreataAccount(String DocID){
+        //- is it a Rider.class or driver.class? or both?
+        Rider riderObj = new Rider();
+        Driver driverObj = new Driver();
+
+        DB.createDriver(DocID, driverObj); //here we created an empty document in the rider or driver collection
```
Comment (evantimms) [inline application/app/src/main/java/com/example/buber/Services/ApplicationService.java]: Just delete the two method calls to create the rider/driver for now, lets keep it as simple as possible and grow the functionality over time.

Comment (evantimms) [reply application/app/src/main/java/com/example/buber/Services/ApplicationService.java]: Return a new User object with the attributes that are passed down from the controller. See the controller code.
image

File: application/app/src/main/java/com/example/buber/Services/ApplicationServiceHelper.java [hunk 16]
```diff
@@ -0,0 +1,9 @@
...
+
+
+    User aftersuccessfulCreataAccount(String DOCID);
```
Comment (evantimms) [inline application/app/src/main/java/com/example/buber/Services/ApplicationServiceHelper.java]: Typo in this and where this is referenced.

**Review by evantimms — CHANGES_REQUESTED** [review]
Hey this looks sweet, lets just do some clean up and remove some excess functionality. Let me know if you have any questions.

File: application/app/src/main/java/com/example/buber/Services/ApplicationService.java (hunk 11 above)
Comment (Madeeha-Anjum) [reply application/app/src/main/java/com/example/buber/Services/ApplicationService.java]: I thought the user was entering a username not an email so in the back end I was appending @uber.com for firebase auth.

File: application/app/src/main/java/com/example/buber/Services/ApplicationServiceHelper.java (hunk 13 above)
Comment (Madeeha-Anjum) [reply application/app/src/main/java/com/example/buber/Services/ApplicationServiceHelper.java]: "pass the reference to the Service class in the callback, good thinking on that one." To do this I needed the interface or it wouldn't work

File: application/app/src/main/java/com/example/buber/Services/ApplicationService.java (hunk 14 above)
Comment (Madeeha-Anjum) [reply application/app/src/main/java/com/example/buber/Services/ApplicationService.java]: "worry about getting a User object"- okay so its a little bit confusing so I'm going to explain what I think and then u can tell me if I'm wrong? 
The user class has attributes like location and account and all that.
(For now, let's assume its a name and a number)
The driver's class extends the user. Rider class extends the user. (this means the driver and rider inherit attributes of the user. )
so when "create account" is called it takes in an email and password and this is stored in the firebase Auth section. 
So now for the collections:
It does not make sense to create a user collection.
it makes more sense to create a rider or driver with the same DocID and then cast them to a user.

File: application/app/src/main/java/com/example/buber/Services/ApplicationService.java (hunk 15 above)
Comment (Madeeha-Anjum) [reply application/app/src/main/java/com/example/buber/Services/ApplicationService.java]: So yeah this makes sense but since Driver and Rider extend user. I might be wrong but in fb we need to create a rider or driver with the same DocID as fbAuth user so that when it returns we can upcast rider or driver to a user

File: application/app/src/main/java/com/example/buber/Services/ApplicationServiceHelper.java (hunk 16 above)
Comment (Madeeha-Anjum) [reply application/app/src/main/java/com/example/buber/Services/ApplicationServiceHelper.java]: ApplicationService.java extends ApplicationServiceHelper

File: application/app/src/main/java/com/example/buber/DB/AuthDBManager.java (hunk 3 above)
Comment (Madeeha-Anjum) [reply application/app/src/main/java/com/example/buber/DB/AuthDBManager.java:83]: It waits yes but there noting telling anything when it is done. so I passed it a method to call when it is done that's what I meant when I said it takes time as in continues to query even after the program continues.

File: application/app/src/main/java/com/example/buber/Services/DatabaseManager.java (hunk 4 above)
Comment (Madeeha-Anjum) [reply application/app/src/main/java/com/example/buber/Services/DatabaseManager.java]: yeah my bad its a duplicate

File: application/app/src/main/java/com/example/buber/Model/Account.java (hunk 9 above)
Comment (Madeeha-Anjum) [reply application/app/src/main/java/com/example/buber/Model/Account.java]: ok

File: application/app/src/main/java/com/example/buber/DB/AuthDBManager.java (hunk 3 above)
Comment (evantimms) [reply application/app/src/main/java/com/example/buber/DB/AuthDBManager.java:83]: Yeah this is fine for now. I have no problems. I think however in the future we should change it so that we just call a static method from the service layer so we are not passing a reference to a whole class.

File: application/app/src/main/java/com/example/buber/Services/ApplicationService.java (hunk 11 above)
Comment (evantimms) [reply application/app/src/main/java/com/example/buber/Services/ApplicationService.java]: for signin we are passing a username and password so thats correct, but appending @Uber.com i don't think is needed. @LukeDelaCruz thoughts?

File: application/app/src/main/java/com/example/buber/Services/ApplicationService.java (hunk 14 above)
Comment (evantimms) [reply application/app/src/main/java/com/example/buber/Services/ApplicationService.java]: I think we are on the same page, but I believe you cannot create multiple documents with the same DocID as the one generated by the FireBaseUser. What I am concerned about is I don't think we all have completely thought over how we want the driver/rider distinction to work so for now (just to get the app auth up and running) I think its better to just have a users collection **for now** and worry about driver rider at a later point after we have given it some thought. It could come down to storing everything in a user collection and casting to Driver/Rider based on how the user is signed in, or have seperate rider/driver collections.

Comment (Madeeha-Anjum) [reply application/app/src/main/java/com/example/buber/Services/ApplicationService.java]: If we want just a user for now, with a name we can also just use "currentuser" in my DBAuth class. you can do "getcurrentUser().getEmail()" or "getcurrentUser().setName()" so this is basically a user object we can pass around. Just a thought.

File: application/app/src/main/java/com/example/buber/Views/Activities/CreateAccountActivity.java [hunk 17]
```diff
@@ -67,22 +70,24 @@ protected void onCreate(Bundle savedInstanceState) {
 
     @Override
     public void onClick(View v) {
-
+        Toast.makeText(this, "Madeeha was here", Toast.LENGTH_LONG).show();
```
Comment (evantimms) [inline application/app/src/main/java/com/example/buber/Views/Activities/CreateAccountActivity.java:73]: Delete

File: application/app/src/main/java/com/example/buber/Services/ApplicationService.java [hunk 18]
```diff
@@ -1,31 +1,45 @@
...
+    // TODO: Lukes code go here
+    private static final String TAG = "ApplicationService";
+
+//   -------------------------------EVAN TODO____________________________________________
     // TODO: In the future lets create a seperate AuthService file, for now this is probably ok
 
     // TODO: Implement
     public static User signIn(String username, String password, User.TYPE type) {
-        if (type == User.TYPE.RIDER) {
+        if (type == User.TYPE.Riders) {
```
Comment (evantimms) [inline application/app/src/main/java/com/example/buber/Services/ApplicationService.java:18]: Shouldn't this be User.TYPE.RIDER? Why did you change this?
